File: src/main/java/com/epri/dss/conversion/impl/PCClassImpl.java
package com.epri.dss.conversion.impl;

import com.epri.dss.common.impl.CktElementClassImpl;
import com.epri.dss.common.impl.DSSClassDefs;
import com.epri.dss.common.impl.DSSGlobals;
import com.epri.dss.conversion.PCClass;
import com.epri.dss.conversion.PCElement;
import com.epri.dss.parser.impl.Parser;

public class PCClassImpl extends CktElementClassImpl implements PCClass {

	int numPCClassProps;

	public PCClassImpl() {
		super();
		this.numPCClassProps = 1;
		this.classType = DSSClassDefs.PC_ELEMENT;
	}

	/**
	 * Add no. of intrinsic properties.
	 */
	protected void countProperties() {
		numProperties = numProperties + numPCClassProps;
		super.countProperties();
	}

	/**
	 * Add properties of this class to propName.
	 *
	 * Define the properties for the base power delivery element class.
	 */
	protected void defineProperties() {
		propertyName[activeProperty + 1] = "spectrum";

		propertyHelp[activeProperty + 1] = "Name of harmonic spectrum for this device.";

		activeProperty = activeProperty + numPCClassProps;

		super.defineProperties();
	}

	protected int classEdit(Object activePCObj, int paramPointer) {
		int result = 0;
		// continue parsing with contents of parser
		if (paramPointer >= 0) {
			PCElement pElem = (PCElement) activePCObj;

			switch (paramPointer) {
			case 1:
				pElem.setSpectrum(Parser.getInstance().makeString());
				break;
			default:
				super.classEdit(activePCObj, paramPointer - numPCClassProps);
				break;
			}
		}
		return result;
	}

	protected void classMakeLike(Object otherObj) {
		PCElement otherPCObj = (PCElement) otherObj;

		PCElement pElem = (PCElement) DSSGlobals.activeDSSObject;

		pElem.setSpectrum(otherPCObj.getSpectrum());
		pElem.setSpectrumObj(otherPCObj.getSpectrumObj());

		super.classMakeLike(otherObj);
	}

}

File: pydss/conversion/PCClassImpl.py
from dss.common.impl.CktElementClassImpl import (CktElementClassImpl,)
from dss.conversion.PCClass import (PCClass,)
from dss.common.impl.DSSGlobals import (DSSGlobals,)
from dss.common.impl.DSSClassDefs import (DSSClassDefs,)
from dss.parser.impl.Parser import (Parser,)


class PCClassImpl(CktElementClassImpl, PCClass):
    numPCClassProps = None

    def __init__(self):
        super(PCClassImpl, self)()
        self.numPCClassProps = 1
        self.classType = DSSClassDefs.PC_ELEMENT

    def countProperties(self):
        """Add no. of intrinsic properties."""
        self.numProperties = self.numProperties + self.numPCClassProps
        super(PCClassImpl, self).countProperties()

    def defineProperties(self):
        """Add properties of this class to propName.

        Define the properties for the base power delivery element class.
        """
        self.propertyName[self.activeProperty + 1] = 'spectrum'
        self.propertyHelp[self.activeProperty + 1] = 'Name of harmonic spectrum for this device.'
        self.activeProperty = self.activeProperty + self.numPCClassProps
        super(PCClassImpl, self).defineProperties()

    def classEdit(self, activePCObj, paramPointer):
        result = 0
        # continue parsing with contents of parser
        if paramPointer >= 0:
            pElem = activePCObj
            _0 = paramPointer
            _1 = False
            while True:
                if _0 == 1:
                    _1 = True
                    pElem.setSpectrum(Parser.getInstance().makeString())
                    break
                if True:
                    _1 = True
                    super(PCClassImpl, self).classEdit(activePCObj, paramPointer - self.numPCClassProps)
                    break
                break
        return result

    def classMakeLike(self, otherObj):
        otherPCObj = otherObj
        pElem = DSSGlobals.activeDSSObject
        pElem.setSpectrum(otherPCObj.getSpectrum())
        pElem.setSpectrumObj(otherPCObj.getSpectrumObj())
        super(PCClassImpl, self).classMakeLike(otherObj)
